v2/datastream-to-spanner/terraform/samples/postgresql-end-to-end/variables.tf

```diff
@@ -0,0 +1,104 @@
+variable "common_params" {
+  description = "Parameters that are common to multiple resources"
+  type = object({
+    project      = string
+    host_project = optional(string)
+    region       = string
+    migration_id = optional(string)
+    # Will be auto-generated if not specified
+    add_policies_to_service_account = optional(bool, true)
+  })
+}
+
+variable "datastream_params" {
+  description = "Parameters to setup Datastream"
+  type = object({
+    private_connectivity = optional(object({
+      private_connectivity_id = optional(string, "priv-conn")
+      vpc_name                = string
+      range                   = string
+    }))
+    private_connectivity_id       = optional(string)
+    source_connection_profile_id  = optional(string, "source-postgresql")
+    postgresql_host               = string
+    postgresql_username           = string
+    postgresql_password           = string
+    postgresql_publication        = string
+    postgresql_replication_slot   = string
+    postgresql_port               = number
+    target_connection_profile_id  = optional(string, "target-gcs")
+    gcs_bucket_name               = optional(string, "live-migration")
+    gcs_root_path                 = optional(string, "/")
+    stream_prefix_path            = optional(string, "data")
+    pubsub_topic_name             = optional(string, "live-migration")
+    stream_id                     = optional(string, "postgresql-stream")
+    max_concurrent_cdc_tasks      = optional(number, 5)
+    max_concurrent_backfill_tasks = optional(number, 20)
+    postgresql_database = object({
+      database = string
+      tables   = optional(list(string))
+    })
+  })
+  validation {
+    condition = (
+      (var.datastream_params.private_connectivity_id == null && var.datastream_params.private_connectivity != null) ||
+      (var.datastream_params.private_connectivity_id != null && var.datastream_params.private_connectivity == null) ||
+      (var.datastream_params.private_connectivity_id == null && var.datastream_params.private_connectivity == null)
+    )
+    error_message = "Exactly one of 'private_connectivity_id' or the 'private_connectivity' block must be provided, not both."
+  }
+}
+
+variable "dataflow_params" {
+  description = "Parameters for the Dataflow job. Please refer to https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/sourcedb-to-spanner/README_Sourcedb_to_Spanner_Flex.md for the description of the parameters below."
```

Comment:
Incorrect doc - should be datastream_to_spanner?

Reply:
This variable is  for google_dataflow_flex_template_job.  Are you suggesting we should change the name of the variable?

Reply:
I mean the doc refers to the sourceDB to Spanner template in the `description` field. You should change this to reference to the datastream to spanner template.

Reply:
Done